src/osp/Active/SysNewton.cpp

```diff
@@ -393,6 +455,45 @@ void SysNewton::body_apply_torque(ACompRigidBody_t &body, Vector3 torque) noexce
     body.m_netTorque += torque;
 }
 
+Vector4 SysNewton::compute_body_CoM(ActiveScene& rScene,
+    ActiveEnt root, Matrix4 currentTransform)
+{
+    ActiveEnt nextChild = root;
+    Vector3 localCoM{0.0f};
+    float localMass = 0.0f;
+
+    while (nextChild != entt::null)
```

Comment:
This should be a for-loop

Reply:
This can't really be done easily due to the way hierarchy nodes are laid out, as far as I can tell

Reply:
Just do

for(ActiveEnt nextChild = root;
    nextChild != entt::null;
    )

to scope the nextChild variable.